diff --git a/linux-rust/src/bluetooth/le.rs b/linux-rust/src/bluetooth/le.rs
index 71a38a8..2f31c09 100644
--- a/linux-rust/src/bluetooth/le.rs
+++ b/linux-rust/src/bluetooth/le.rs
@@ -106,10 +106,7 @@ pub async fn start_le_monitor(tray_handle: Option<Handle<MyTray>>) -> bluer::Resu
                         && irk_bytes.len() == 16
                     {
                         let irk: [u8; 16] = irk_bytes.as_slice().try_into().unwrap();
-                        debug!(
-                            "Verifying RPA {} for airpods MAC {} with IRK {}",
-                            addr_str, airpods_mac, info.le_keys.irk
-                        );
+                        debug!("Verifying RPA {} for AirPods MAC {}", addr_str, airpods_mac);
                         if verify_rpa(&addr_str, &irk) {
                             info!(
                                 "Matched our device ({}) with the irk for {}",
